src/core/qgsvectorlayer.cpp

```diff
@@ -4415,13 +4421,29 @@ void QgsVectorLayer::invalidateSymbolCountedFlag()
 
 void QgsVectorLayer::onFeatureCounterCompleted()
 {
+  doneTask();
   onSymbolsCounted();
-  mFeatureCounter = nullptr;
 }
 
 void QgsVectorLayer::onFeatureCounterTerminated()
 {
-  mFeatureCounter = nullptr;
+  doneTask();
+}
+
+void QgsVectorLayer::doneTask()
```

Comment:
Signal to indicate a task is done, currently used to keep track of all finished counting.